src/Request/RequestFactoryInterface.php

```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace Psr\Http\Factory;
+namespace Psr\Http\Factory\Request;
```

Comment:
I don't see the need to "group" single classes, those extra namespaces are redundant. I would group together the Request factories, and leave the URI and uploaded file factories in the root namespace.

```
Request/RequestFactoryInterface.php
Request/ServerRequestFactoryInterface.php
Request/GlobalServerRequestFactoryInterface.php
Response/ResponseFactoryInterface.php
Stream/CallbackStreamFactoryInterface.php
Stream/ResourceStreamFactoryInterface.php
Stream/StreamFactoryInterface.php
Stream/StringStreamFactoryInterface.php
UploadedFileFactoryInterface.php
UriFactoryInterface.php
```

Or even just group the streams and leave everything else.

```
RequestFactoryInterface.php
ServerRequestFactoryInterface.php
GlobalServerRequestFactoryInterface.php
ResponseFactoryInterface.php
UploadedFileFactoryInterface.php
UriFactoryInterface.php
Stream/CallbackStreamFactoryInterface.php
Stream/ResourceStreamFactoryInterface.php
Stream/StreamFactoryInterface.php
Stream/StringStreamFactoryInterface.php
```

Dedicated namespaces for individual factories seems a little overkill.